### PR TITLE
Remove need to build Open CV to build/use the Object Rec Connector

### LINKED ISSUE #287: ObjRec: Remove need to build OpenCV.

### PR DESCRIPTION
**Pull request by fhcarter**
Fixes #287 

(Apologies for the size of the PR.  There's really no way to do this incrementally.)

This PR substantially changes the way the Object Recognition Connector is built.  Previously, builders needed to have downloaded & built OpenCV -- which was sometimes no mean feat.  This PR changes the base from OpenCV to JavaCV, a more packaged solution that includes all of OpenCV (the same underlying technology is used).  This should make it easier to build the OR connector.

Additionally, some tests have been added & others cleaned up.  A fair amount of code cleanup was performed as well.

### CODE REVIEW

**Review by AlexBlumer — APPROVED** [review]
LGTM, just a few nits

**Comment by jcguion** [comment]
nit - is the comma position intentional (vs. previous line)

```
                , (after - before) / 1000, String.format("%03d", (after - before) % 1000));
```
								
[objectRecognitionSource/src/main/java/io/vantiq/extsrc/objectRecognition/imageRetriever/NetworkStreamRetriever.java](https://github.com/vantiq/vantiq-extension-sources/blob/472ce0a090d627872897887a1e9ea9337ceda107/objectRecognitionSource/src/main/java/io/vantiq/extsrc/objectRecognition/imageRetriever/NetworkStreamRetriever.java#L190) (Line 190)

**Review by jcguion — APPROVED** [review]
LGTM - minor feedback/nits

File: objectRecognitionSource/src/main/java/io/vantiq/extsrc/objectRecognition/imageRetriever/RetrieverBase.java [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+        byte[] imageBytes = null;
+        if (image.empty()) {
+            log.warn("Cannot convert empty image to jpg");
+        } else if (imencode(".jpg", image, bytes)) {
+            if (log.isDebugEnabled()) {
+                log.debug("bytes stuff: limit: {}, position: {}, capacity: {}", bytes.limit(),
+                        bytes.position(), bytes.capacity());
+            }
+            imageBytes = bytes.getStringBytes();
+            log.debug("JPG length is: {}", imageBytes.length);
```
[the remark this answers is not in the document]
Comment (fhcarter) [reply objectRecognitionSource/src/main/java/io/vantiq/extsrc/objectRecognition/imageRetriever/RetrieverBase.java:67]: Most were from Sider complaints -- it seems to complain if there's a method invocation vs. just passing a property.  But added it here for consistency.

File: objectRecognitionSource/src/test/java/io/vantiq/extsrc/objectRecognition/neuralNet/TestNoProcessorQueries.java [hunk 2]
```diff
@@ -237,6 +255,8 @@ public void testProcessNextFrameBoth() throws InterruptedException {
         
         // Check there is only one file, and it's name is equivalent to the last saved file
         outputDirFiles = outputDir.listFiles();
+        assert outputDirFiles != null;
+        assert outputDirFiles != null;
```
[the remark this answers is not in the document]
Comment (fhcarter) [reply objectRecognitionSource/src/test/java/io/vantiq/extsrc/objectRecognition/neuralNet/TestNoProcessorQueries.java:259]: Foghorn-leghorn testing fixed...

File: objectRecognitionSource/build.gradle [hunk 3]
```diff
@@ -108,17 +70,13 @@ applicationDistribution.from("src/main/resources") {
 
 test {
     // set heap size for the test JVM(s)
+    // Java CV seems to impart more memory requirements for getting the same test results
+    // with gradle that we do with the IDE (even though the IDE is running tests via gradle).
     minHeapSize = "2048m"
     maxHeapSize = "2048m"
```
[the remark this answers is not in the document]
Comment (fhcarter) [reply objectRecognitionSource/build.gradle:76]: I think it happened for two reasons, one of which went in while this branch was being worked on.  Removed the comment as it's not quite accurate.

**Comment by fhcarter** [comment]
(Note -- this will probably sit & wait for some verification from those in the field. It's a big enough change that I would like a little external verification.)

**Pull request merged**